core/src/test/java/org/apache/iceberg/TestTransaction.java

```diff
@@ -699,4 +700,23 @@ public void testTransactionRewriteManifestsAppendedDirectly() throws IOException
 
     Assert.assertFalse("Append manifest should be deleted on expiry", new File(newManifest.path()).exists());
   }
+
+  @Test
+  public void testSimpleTransactionNotDeletingMetadataOnUnknownSate() throws IOException {
+    Table table = TestTables.tableWithCommitSucceedButStateUnknown(tableDir, "test");
+
+    Transaction transaction = table.newTransaction();
+    transaction.newAppend()
+        .appendFile(FILE_A)
+        .commit();
+    AssertHelpers.assertThrows("Transaction commit should fail with CommitStateUnknownException",
+                               CommitStateUnknownException.class, "datacenter on fire",
+                               () -> transaction.commitTransaction());
```

Comment:
The continuation indent should be 4 spaces here.

Reply:
That conflicts with method parameter alignment

Reply:
Method arguments are optionally aligned with the start of the first arg in method definitions. Method calls always use 4 spaces for continuation indents.